src/asm/symbol.c

```diff
@@ -544,12 +542,21 @@ struct Symbol *sym_AddLocalLabel(char const *symName)
 	if (strchr(localName + 1, '.'))
 		fatalerror("'%s' is a nonsensical reference to a nested local label\n",
 			   symName);
+	// If no label scope, check that the label is already fully qualified
+	if (!labelScope && !ALLOW_LOCAL_LABEL_ANYWHERE) {
+		error("Local label '%s' in main scope\n", symName);
+		return NULL;
+	}
 
 	if (localName == symName) {
+		if (!labelScope) {
+			error("Relative local label '%s' in main scope\n", symName);
```

Comment:
I feel like "unqualified" would make more sense then "relative"? The latter evokes address more than name, imo.